### ISSUE TITLE
binary_cfdr fails completely if input is not sorted by chr

### ISSUE THREAD

**Issue by jan-glx** [annahutch/fcfdr#6]
``` r
# remotes::install_github("annahutch/fcfdr")
library(fcfdr)
library(ggplot2)

N <- 1000
p <- runif(N)

q <- rnorm(N)
chr <- rep(c("a", "b"), N/2)
res <- fcfdr::binary_cfdr(p, q>0, chr)
ggplot(res, aes(x=p, y=v, color=q)) + geom_point() + scale_x_log10() + scale_y_log10() + geom_abline()
```

![](https://i.imgur.com/ef3Femo.png)

``` r
chr <- rep(c("a", "b"), each = N/2)
res <- fcfdr::binary_cfdr(p, q>0, chr)
ggplot(res, aes(x=p, y=v, color=q)) + geom_point() + scale_x_log10() + scale_y_log10() + geom_abline()
```

![](https://i.imgur.com/VuC6h82.png)

<sup>Created on 2021-04-30 by the [reprex package](https://reprex.tidyverse.org) (v1.0.0)</sup>
Also, I would expect all points to lie on one of 4 lines?

**Comment by annahutch** [comment]
Hi, yes there is currently a bug with the binary cFDR code which I haven't yet gotten round to fixing (hence I removed the relevant binary cFDR vignette from the software webpage for the time being). I hope to get round to this in the next couple of weeks and will get back to you then. Many thanks. Anna

**Comment by annahutch** [comment]
Hi again. In terms of the issue relating to the ordering of SNPs with respect to their chromosome, this has now been fixed (it was a simple error in using `unlist()` rather than `unsplit()`). 

In terms of your other issue relating to the output, we have noted this and will take a look at it in due course (probably whilst we prepare the manuscript relating to the binary mode of Flexible cFDR). I have also added an "under development" warning to the `binary_cfdr()` code so that it is clearer to users that this function is still under development. Thanks again, Anna.
